### PR TITLE
Unirule rest for get by unirule id

### PR DESCRIPTION
**Pull request by ahmadshadab**

### CODE REVIEW

File: core-parser/src/main/java/org/uniprot/core/parser/tsv/unirule/UniRuleEntryValueMapper.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+    private StringBuilder getProteinName(List<Annotation> annotations) {
+        ProteinDescription proteinDescription =
+                annotations.stream()
+                        .map(Annotation::getProteinDescription)
+                        .findFirst()
+                        .orElse(null);
+        StringBuilder builder = new StringBuilder();
+        if (Utils.notNull(proteinDescription)) {
+            String fullName = proteinDescription.getRecommendedName().getFullName().getValue();
```
Comment (LeonardoGonzales) [inline core-parser/src/main/java/org/uniprot/core/parser/tsv/unirule/UniRuleEntryValueMapper.java:160]: Required: I think that in this code we might get a nullpointerException if the description do not have a recommended name. For example: this accession do not have recommended name: https://www.uniprot.org/uniprot/A0A514LIK3.txt, Can you also please add a test for this use case?

File: core-parser/src/test/java/org/uniprot/core/parser/tsv/unirule/UniRuleEntryValueMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import org.uniprot.core.unirule.impl.ConditionBuilder;
+import org.uniprot.core.unirule.impl.ConditionSetBuilder;
+import org.uniprot.core.unirule.impl.ConditionValueBuilder;
+import org.uniprot.core.unirule.impl.InformationBuilder;
+import org.uniprot.core.unirule.impl.RuleBuilder;
+import org.uniprot.core.unirule.impl.RuleBuilderTest;
+import org.uniprot.core.unirule.impl.UniRuleEntryBuilder;
+import org.uniprot.core.unirule.impl.UniRuleIdBuilder;
+
+class UniRuleEntryValueMapperTest {
```
Comment (LeonardoGonzales) [inline core-parser/src/test/java/org/uniprot/core/parser/tsv/unirule/UniRuleEntryValueMapperTest.java:32]: Excellent coverage!!!

**Review by LeonardoGonzales — COMMENTED** [review]
There is just one comment, that I would you like you to have an attention. All the others are optionals (cosmetics)

File: core-parser/src/main/java/org/uniprot/core/parser/tsv/unirule/UniRuleEntryValueMapper.java (hunk 1 above)
Comment (ahmadshadab) [reply core-parser/src/main/java/org/uniprot/core/parser/tsv/unirule/UniRuleEntryValueMapper.java:160]: Thanks for this finding. 👍

File: core-parser/src/main/java/org/uniprot/core/parser/tsv/unirule/UniRuleEntryValueMapper.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    private String getTaxonomicScope(UniRuleEntry uniRuleEntry) {
+        Rule mainRule = uniRuleEntry.getMainRule();
+        Optional<Condition> optCondition =
+                mainRule.getConditionSets().stream()
+                        .map(ConditionSet::getConditions)
+                        .flatMap(Collection::stream)
+                        .filter(condition -> "taxon".equals(condition.getType()))
+                        .findFirst();
+        if (optCondition.isPresent()) {
```
[the remark this answers is not in the document]
Comment (ahmadshadab) [reply core-parser/src/main/java/org/uniprot/core/parser/tsv/unirule/UniRuleEntryValueMapper.java:65]: Added. thanks

**Pull request merged**